xclim/indices/_agro.py

```diff
@@ -1680,4 +1684,7 @@ def chill_units(tas: xarray.DataArray, freq: str = "YS") -> xarray.DataArray:
         ),
     )
     cu = cu.where(tas.notnull())
-    return cu.resample(time=freq).sum().assign_attrs(units="")
+    daily = cu.resample(time="1D").sum()
+    if positive_only:
+        daily = daily.where(daily > 0)
+    return daily.resample(time=freq).sum().assign_attrs(units="")
```

Comment:
```suggestion
    if positive_only:
        daily = cu.resample(time="1D").sum()
        cu = daily.where(daily > 0)
    return cu.resample(time=freq).sum().assign_attrs(units="")
```
This way we can avoid adding operations when not needed?